### ISSUE TITLE
FileSequence fails to parse date-time file name

### ISSUE THREAD

**Issue by jonaskluger** [justinfx/fileseq#117]
The following code breaks for movie files with a date/time file naming.
Yes, these are not typical VFX file names. It is for files in a folder (screenrecordings from OBS)

The `findSequencesOnDisk` detects movie files in the folder and returns them like this: `C:\foo\bar\2023-01-14_12-32-30@@@.mov`

```
# Example code
from fileseq import FileSequence, findSequencesOnDisk

path_directory = '/foo/bar'
sequences = findSequencesOnDisk(path_directory, strictPadding=True)
for sequence in sequences:
	foo = FileSequence(sequence)
```

```
Traceback (most recent call last):
  File "C:\foobar\venv\lib\site-packages\fileseq\filesequence.py", line 58, in __init__
    self._frameSet = FrameSet(frames)
  File "C:\foobar\venv\lib\site-packages\fileseq\frameset.py", line 169, in __init__
    start, end, modifier, chunk = FrameSet._parse_frange_part(part)
  File "C:\foobar\venv\lib\site-packages\fileseq\frameset.py", line 1049, in _parse_frange_part
    raise ParseException(msg.format(frange, cls.FRANGE_RE.pattern))
fileseq.exceptions.ParseException: Could not parse "12-32-30": did not match
    ^
    (-?\d+)             # start frame
    (?:                 # optional range
        -               #   range delimiter
        (-?\d+)         #   end frame
        (?:             #   optional stepping
            ([:xy]{1})  #     step format
            (-?\d+)     #     step value
        )?
    )?
    $


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".\test.py", line 51, in <module>
    for foo in get_paths_from_disk(path, True):
  File "C:\foobar\src\app\extended\pathes.py", line 174, in get_paths_from_disk
    result += list(FileSequence(sequence))
  File "C:\foobar\venv\lib\site-packages\fileseq\filesequence.py", line 64, in __init__
    raise ParseException(msg.format(sequence))
fileseq.exceptions.ParseException: Failed to parse FileSequence: C:\foo\bar\2023-01-14_12-32-30@@@.mov
```

**Comment by justinfx** [comment]
Is your main concern reporting the fact that `findSequencesOnDisk()` is returning the movie file as a sequence, or that it results in an exception later in the `FileSequence` constructor?

I am confused about what exactly you are reporting, because once you ask `findSequencesOnDisk()` for a list of `FileSequence` results, you then pass each one back through a `FileSequence(seq)` to construct another instance? I can see how that would lead to problems.

`findSequencesOnDisk` starts with a filesystem directory and scans all of the individual file names, parsing them with the [DISK_PATTERN](https://github.com/justinfx/fileseq/blob/v1.15.0/src/fileseq/constants.py#L156) in order to extract the dirname, basename, frame, and extension. In the case of your movie file, it thinks the frame is "-30", which I think makes perfect sense given the library doesn't have a way to know its mean to be a date string for a single movie file. 

Then you take that resulting `<FileSequence: '/tmp/seqtest/2023-01-14_12-32-30@@@.mov'>` and you pass it back to the `FileSequence` constructor, which is going to take the string "/tmp/seqtest/2023-01-14_12-32-30@@@.mov" and put it through the [SPLIT_PATTERN](https://github.com/justinfx/fileseq/blob/v1.15.0/src/fileseq/constants.py#L121) in order to extract the dirname, basename, frame range, padding chars, and ext. Here is where we fork some outcomes...

1) In your case, given the line that crashed in your traceback, you seem to be using at best fileseq v1.10.0 that was released almost 3 years ago. This does indeed crash while trying to parse.

2) If you were to try the recent release, such as v1.15.0, it would indeed succeed in reparsing the sequence string. Because this is already a sequence string, its going to think the range is "-32-30", which obviously has no relation to the original sequence object. 

So can you provide some details on what exactly you would like to see improved? And is there a reason you aren't testing this against a more recent release of fileseq?

**Comment by jonaskluger** [comment]
Hey Justin,
thanks a lot for the detailed explanation. The issue relates to the Exception of FileSequence in version 1.10.0

**TL;DR**
Upgrading to the lastes version (1.15.0) did fix the parsing exception of movie files with this date/time naming.


`findSequencesOnDisk ` works correctly. It detects, as expected, multiple movie files with a sequential naming and returns them as a `FileSequence`.

As you pointed out in version 1.10.0 the `<FileSequence: '/tmp/seqtest/2023-01-14_12-32-30@@@.mov'>` did fail to parse.
In version 1.15.0 it works.
Passing a `FileSequence` into another `FileSequence` was unnecessary and failed. User mistake on my end.

Why I didn't upgrade - it's a very complex code setup for IO that can deal with all different sorts of file(s) and sequences. At this step the incoming files are not yet in an organized format/pipeline, that's why it has to deal with the weirdest file names. Didn't upgrade the version for so long because everything else worked perfectly. Never touch a running system.

The issue is now resolved. Please feel free to close this issue.
Thanks a lot for creating this amazing repo! I tip my hat to you!

**Comment by justinfx** [comment]
That's great to hear that upgrading to the newer version solve the problem for you! Easy fix!
I understand the desire to avoid upgrading versions when something is working. We do try our best to avoid breaking existing behaviour between minor releases.
